server/webapp/WEB-INF/rails.new/app/assets/javascripts/pipeline_configs/models/materials.js

```diff
@@ -69,12 +69,11 @@ define(['mithril', 'lodash', 'string-plus', './model_mixins', './encrypted_value
   Materials.Material = function (type, hasFilter, data) {
     this.constructor.modelType = 'material';
     Mixins.HasUUID.call(this);
-
-    this.parent = Mixins.GetterSetter();
-
-    this.type       = m.prop(type);
-    this.name       = m.prop(s.defaultToIfBlank(data.name, ''));
-    this.autoUpdate = m.prop(data.autoUpdate);
+    this.parent           = Mixins.GetterSetter();
+    this.type             = m.prop(type);
+    this.name             = m.prop(s.defaultToIfBlank(data.name, ''));
+    this.autoUpdate       = m.prop(data.autoUpdate);
+    this.testConnectionVM = m.prop(new Materials.Material.vm.testConnection());
 
```

Comment:
After a long discussion, we've concluded that this view model should be pulled out of materials because it's better for testability and re-usability to render check connection as a separate component. The state must not be maintained in the actual material as testing a connection is a view only concern.